### PR TITLE
Updated DG

### PR DESCRIPTION
**Pull request by YHWong20**
## Changes made
* Added overall architecture diagram
* Updated User stories
* Added some manual test cases
* Reorganized some sections

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -289,10 +294,26 @@ Small business (retail shop) owners who:
 
 ## User Stories
 
-| Version | As a ... | I want to ...             | So that I can ...                                           |
-| ------- | -------- | ------------------------- | ----------------------------------------------------------- |
-| v1.0    | new user | see usage instructions    | refer to them when I forget how to use the application      |
-| v2.0    | user     | find a to-do item by name | locate a to-do without having to go through the entire list |
+| Version | As a ...   | I want to ...                                                            | So that I can ...                                                                 |
```
Comment (PureUsagi) [inline docs/DeveloperGuide.md:297]: Thanks for updating user stories!

File: docs/images/Architecture.png
Comment (PureUsagi) [inline docs/images/Architecture.png:1]: Should we add the logger class as well?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -302,8 +323,100 @@ Small business (retail shop) owners who:
...
 ## Glossary
 
-* **CLI** - Command-line Interface
+* **CLI** - Command Line Interface
+* **GUI** - Graphical User Interface
 
 ## Instructions for manual testing
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+> Testers are encouraged to do more exploratory testing with the following commands.
```
Comment (PureUsagi) [inline docs/DeveloperGuide.md:331]: Should we also let them know that they can access the logs and data file directly for manual testing?

**Review by PureUsagi — APPROVED** [review]
Overall great work on the additions to the DG!

File: docs/images/Architecture.png
Comment (YHWong20) [reply docs/images/Architecture.png:1]: I'll do that, thanks!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (YHWong20) [reply docs/DeveloperGuide.md:331]: Sure, I'll add that in.

**Pull request merged**